yew-ui/src/components/attendants.rs

```diff
@@ -72,43 +71,72 @@ pub struct AttendantsComponentProps {
 pub struct AttendantsComponent {
     pub connection: Option<Connection>,
     pub peer_decode_manager: PeerDecodeManager,
+    pub media_device_access: MediaDeviceAccess,
     pub outbound_audio_buffer: [u8; 2000],
     pub share_screen: bool,
     pub webtransport_enabled: bool,
     pub mic_enabled: bool,
     pub video_enabled: bool,
     pub error: Option<String>,
-    pub media_access_granted: bool,
 }
 
-impl Component for AttendantsComponent {
-    type Message = Msg;
-    type Properties = AttendantsComponentProps;
+impl AttendantsComponent {
+    fn is_connected(&self) -> bool {
+        match &self.connection {
+            Some(connection) => connection.is_connected(),
+            None => false,
+        }
+    }
 
-    fn create(ctx: &Context<Self>) -> Self {
-        let webtransport_enabled = ctx.props().webtransport_enabled;
+    fn create_peer_decoder_manager(ctx: &Context<Self>) -> PeerDecodeManager {
         let mut peer_decode_manager = PeerDecodeManager::new();
-        let link = ctx.link().clone();
-        peer_decode_manager.on_peer_added = Callback::from(move |email| {
-            link.send_message(Msg::OnPeerAdded(email));
-        });
-        let link = ctx.link().clone();
-        peer_decode_manager.on_first_frame = Callback::from(move |(email, media_type)| {
-            link.send_message(Msg::OnFirstFrame((email, media_type)));
-        });
+        peer_decode_manager.on_peer_added = {
+            let link = ctx.link().clone();
+            Callback::from(move |email| link.send_message(Msg::OnPeerAdded(email)))
+        };
+        peer_decode_manager.on_first_frame = {
+            let link = ctx.link().clone();
+            Callback::from(move |(email, media_type)| {
+                link.send_message(Msg::OnFirstFrame((email, media_type)))
+            })
+        };
         peer_decode_manager.get_video_canvas_id = Callback::from(|email| email);
         peer_decode_manager.get_screen_canvas_id =
             Callback::from(|email| format!("screen-share-{}", &email));
+        peer_decode_manager
+    }
+
+    fn create_media_device_access(ctx: &Context<Self>) -> MediaDeviceAccess {
+        let mut media_device_access = MediaDeviceAccess::new();
+        media_device_access.on_granted = {
+            let link = ctx.link().clone();
```

Comment:
@ronen seems like this callback is getting called even if you refuse access to the camera.